staging_service/autodetect/GenerateMappings.py

```diff
@@ -0,0 +1,303 @@
+"""
+* This script generates various potential api responses and serves as a list of supported apps and extensions
+for the staging service endpoint .
+* Afterwards, we can pick a json or yaml file and edit those, or keep editing this file in the future to generate the mappings
+
+
+================================
+# In Scope Importer Apps
+* Note: FastQ Interleaved/Uninterleaved custom UIs not yet implemented
+* Note: MergeMetabolicAnnotations and PredictPhenotypes also require an object in addition to a file
+
+================================
+# Out of scope importer notes
+* Note: MultipleSequenceAlignmentApp Commented out because it has a weird interface
+
+================================
+# Batch App Importer notes
+* Commented out all batch apps, as those take a directory from the narrative itself
+
+Functionality: Running this script will
+* Save this indirectly as a json file
+* Save this indirectly as a yaml file
+
+* Note: We should serve the generated content from memory
+* Note: This doesn't handle if we want to have different output types based on file extensions feeding into the same app
+"""
+import copy
+from collections import defaultdict, OrderedDict
+
+from staging_service.autodetect.Mappings import *
+
+mapping = defaultdict(list)
+
+mapping[SRA] = [
+    {
+        "title": "SRA Reads",
+        "app": "kb_uploadmethods/import_fastq_sra_as_reads_from_staging",
+        "output_type": ["KBaseFile.SingleEndLibrary", "KBaseFile.PairedEndLibrary"],
+    }
+]
+
+mapping[FASTQ] = [
+    {
+        "title": "FastQ Reads Interleaved",
+        "app": "kb_uploadmethods/import_fastq_sra_as_reads_from_staging",
+        "output_type": ["KBaseFile.SingleEndLibrary", "KBaseFile.PairedEndLibrary"],
+    },
+    {
+        "title": "FastQ Reads UnInterleaved",
+        "app": "kb_uploadmethods/import_fastq_sra_as_reads_from_staging",
+        "output_type": ["KBaseFile.SingleEndLibrary", "KBaseFile.PairedEndLibrary"],
+    },
+]
+
+mapping[FASTA] = [
+    {
+        "title": "Assembly",
+        "app": "kb_uploadmethods/import_fasta_as_assembly_from_staging",
+        "output_type": ["KBaseGenomeAnnotations.Assembly"],
+    },
+    # {
```

Comment:
having the comment about why the app has been commented out would be more helpful

Reply:
I have the comment at the top of the file: All batch apps have been commented out

Reply:
It'd be helpful if you could copy it here so that devs don't have to scroll back up to the top of the file and check why it is each of these entries are commented out.

Reply:
ok done